plugin/pkg/scheduler/schedulercache/interface.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package schedulercache
+
+import "k8s.io/kubernetes/pkg/api"
+
+// State Machine (life cycle) of a pod in scheduler's cache:
+//
+//           Assume            ConfirmScheduled
+// Initial +--------> Binded  +----------------> Confirmed
+//                      +                         +
+//                      |                         |
+//                      |                         | Remove
+//                      | Remove                  |
+//                      |                         v
+//                      +-------------------->  Deleted (End)
+//
+// Note:
+// - Both "Initial" and "Deleted" pods do not actually exist in cache.
+// - Binded -> Confirmed:
+//   - Assumed pod might have been expired and deleted.
+// - Binded -> Deleted: Missing Add event
+//   - Assumed pod might have been expired and deleted.
+
+// NodeInfoCache
+type NodeInfoCache interface {
+	// AssumePod
```

Comment:
AssumbPod assumes a pod to be scheduled. The implementation might decide the policy to remove the assumed pod before it is confirmed to be scheduled.